runtime/str.go

```diff
@@ -205,6 +205,37 @@ func strEq(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return strCompare(v, w, False, True, False), nil
 }
 
+func strFind(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	// TODO: Support unicode sub string
+	expectedTypes := []*Type{StrType, StrType, IntType, IntType}
+	argc := len(args)
+	if argc == 2 || argc == 3 {
+		expectedTypes = expectedTypes[:argc]
+	}
+	if raised := checkMethodArgs(f, "find/index", args, expectedTypes...); raised != nil {
+		return nil, raised
+	}
+	s := toStrUnsafe(args[0]).Value()
+	l := len(s)
+	start, end := 0, l
+	if argc >= 3 {
+		start = adjustIndex(toIntUnsafe(args[2]).Value(), l)
+	}
+	if argc == 4 {
+		end = adjustIndex(toIntUnsafe(args[3]).Value(), l)
+	}
+	if start > end || start > l {
+		return NewInt(-1).ToObject(), nil
+	}
+	sep := toStrUnsafe(args[1]).Value()
+	s = s[start:end]
```

Comment:
If the user passes end > len(s), I think this will panic. The nice thing about clamping start and end to 0 <= start/end <= len(s) is that it's guaranteed to be safe to pass into slice bounds. You need to be able to determine `start > l` so I would suggest that you store `toIntUnsafe(args[2]).Value()` as a separate value (e.g. startRaw) and check `startRaw > l` to address the bug you found, but then calculate `start = adjustIndex(startRaw, l)`